### PR TITLE
Uml 2674

### PR DESCRIPTION
**Pull request by jamesrwarren**
# Purpose

Introduced a lambda function to upload metrics in stats table

Fixes UML-2674

## Approach

- Modified the script to get data in correct format

- Build a lambda image using a  docker file and introduced docker-compose entry for local

    Tested using curl in local

- Circle CI changes to build the image and push it to ECR

- Created the lambda in terraform and provided roles/permissions etc

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Use a Lasting Power of Attorney service_

## Checklist

* [X] I have performed a self-review of my own code
* [X] I have added relevant logging with appropriate levels to my code
* ~New event_codes have been documented on the [wiki page](https://opgtransform.atlassian.net/wiki/spaces/LSML2/pages/3277881441/Understanding+the+event+logs)~
* [X] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* ~I have added tests to prove my work~
* ~I have added welsh translation tags and updated translation files~
* ~I have run an accessibility tool on any pages I have made changes to and fixed any issues found~
*~I have notified the Interaction Designer of any content changes so that appropriate screenshots/flow diagram changes can be made~
* [ ] The product team have tested these changes

### CODE REVIEW

File: docker-compose.dependencies.yml [hunk 1]
```diff
@@ -64,9 +64,9 @@ services:
   data-lpa:
     container_name: data-lpa
     image: stoplight/prism:4
-    build: # Does not have M1 native / arm64 images as yet but builds fine.
```
Comment (lukejolliffe) [inline docker-compose.dependencies.yml:67]: Should we commit this or leave it as a local change?

File: docker-compose.yml [hunk 2]
```diff
@@ -247,5 +248,27 @@ services:
...
+  upload-stats-lambda:
+    container_name: upload-statistics
+    image: upload-statistics
+    build:
+      context: ./lambda-functions/upload-statistics
+      dockerfile: Dockerfile
+    environment:
+      ENVIRONMENT: local
+      REGION: eu-west-1
+      AWS_ACCESS_KEY_ID: 'dummy'
```
Comment (lukejolliffe) [inline docker-compose.yml:267]:
```suggestion
      AWS_ACCESS_KEY_ID: '-'
```

Comment (lukejolliffe) [reply docker-compose.yml:267]: Just to keep consistent with other keys in this file

File: docker-compose.yml [hunk 3]
```diff
@@ -247,5 +248,27 @@ services:
...
+      AWS_SECRET_ACCESS_KEY: 'dummy'
```
[the remark this answers is not in the document]
Comment (lukejolliffe) [reply docker-compose.yml:268]: again to keep consistent

File: lambda-functions/upload-statistics/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+curl -XPOST "http://localhost:9007/2015-03-31/functions/function/invocations" -d '{}'
+```
+
+# Restart lambda on code changes
+
+Run following to restart the lambda when making code changes as lambda runs in runtime environment and needs 
+bringing up again for changes to take effect.
+
+```
+docker-compose -f docker-compose.yml -f docker-compose.dependencies.yml up -d upload-stats-lambda
```
Comment (lukejolliffe) [inline lambda-functions/upload-statistics/README.md:32]:
```suggestion
make up upload-stats-lambda
```

Comment (lukejolliffe) [reply lambda-functions/upload-statistics/README.md:32]: this should also work and is easier

File: lambda-functions/upload-statistics/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+# To view logs
+
+```
+docker-compose -f docker-compose.yml -f docker-compose.dependencies.yml logs -f upload-stats-lambda
```
Comment (lukejolliffe) [inline lambda-functions/upload-statistics/README.md:38]:
```suggestion
make logs upload-stats-lambda
```

Comment (lukejolliffe) [reply lambda-functions/upload-statistics/README.md:38]: this should also work and is easier

File: docker-compose.dependencies.yml (hunk 1 above)
Comment (SeemaMenon) [reply docker-compose.dependencies.yml:67]: Oh yeah, I think we leave it as a local change for now. 
Deleting this file here.

Comment (jamesrwarren) [reply docker-compose.dependencies.yml:67]: I was happy to commit this as we haven't fixed the root cause and most of us don't have the m1 macs but we can revert

**Review by lukejolliffe — APPROVED** [review]

**Pull request merged**